core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -166,7 +179,34 @@ class DefaultAutoTopicCreationManager(
         debug(s"Auto topic creation completed for ${creatableTopics.keys}.")
         clearInflightRequests(creatableTopics)
       }
-    })
+    }
+
+    val channelManager = this.channelManager.getOrElse {
+      throw new IllegalStateException("Channel manager must be defined in order to send CreateTopic requests.")
+    }
+
+    metadataRequestContext match {
+      case Some(context) =>
+        val requestVersion =
+          channelManager.controllerApiVersions() match {
+            case None =>
+              ApiKeys.CREATE_TOPICS.latestVersion()
```

Comment:
This is interesting. I guess the situation is when we have not connected to the controller yet. What happens if the controller does not support this version? It should send back UNSUPPORTED_VERSION in the envelope response I think, right? Should we retry the request or do we just rely on the client to resend the Metadata request so that we can retry?

Reply:
I guess we could rely on client to retry Metadata request for simplicity.